testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -193,19 +196,282 @@ def process_test_clean(test_tag, work_dir):  # {{{
+    local_parallel_code += "end_time = time.time()\n"
+    local_parallel_code += "print('parallel run time: {} min'.format((end_time - start_time) / 60))\n"
+    local_parallel_code = write_regression_script_testcase_data_bottom(
+        local_parallel_code)
+
+    return local_parallel_code
+    # }}}
+
+
+def write_regression_local_parallel_top(work_dir, suite_tag, nodes, data_dic):
+    # {{{
+    local_parallel_code = "#!/usr/bin/env python\n\n\n"
+    local_parallel_code += "import time\n"
+    local_parallel_code += "import psutil\n"
+    local_parallel_code += "import sys\n"
+    local_parallel_code += "import os\n"
+    local_parallel_code += "import subprocess\n"
+    local_parallel_code += "import numpy as np\n"
+    local_parallel_code += "import multiprocessing as mp\n"
+    local_parallel_code += "out, err = subprocess.Popen(['nproc'],stdout=subprocess.PIPE, stderr=subprocess.STDOUT).communicate()\n"
+    local_parallel_code += "nodes = {}\n".format(nodes)
+    local_parallel_code += "number_of_procs = nodes * int(out.split()[0])\n"
+    local_parallel_code += "print('Number of usable Processors: {}'.format(number_of_procs))\n"
+    local_parallel_code += "os.environ['PYTHONUNBUFFERED'] = '1'\n"
+    local_parallel_code += "test_failed=False\n"
+    local_parallel_code += "if not os.path.exists('case_outputs'):\n"
+    local_parallel_code += "    os.makedirs('case_outputs')\n"
+    local_parallel_code += "base_path = '" + work_dir + "'\n"
+    local_parallel_code += "os.chdir(base_path)\n"
+
+    for child in suite_tag:
+        for script in child:
+            script_name = script.attrib['name']
+            data_dic['run_file_names'].append(script_name)
+
+    local_parallel_code += "\n\n\n"
+    return local_parallel_code
+    # }}}
+
+
+def write_regression_script_testcase_data_bottom(regression_script_code):
+    # {{{
+    regression_script_code += "print('\\n\\n\\nTEST RUNTIMES & ERROR:')\n"
+    regression_script_code += "case_output = '/case_outputs/'\n"
+    regression_script_code += "totaltime = 0\n"
+    regression_script_code += "for _, _, files in os.walk(base_path + case_output):\n"
+    regression_script_code += "    for afile in sorted(files):\n"
+    regression_script_code += "        outputfile = base_path + case_output + afile\n"
+    regression_script_code += "        runtime = np.ceil(float(subprocess. check_output(\n  ['grep', 'real', outputfile])."
+    regression_script_code += "decode('utf-8').split('\\n')[-2].split()[1]))\n"
+    regression_script_code += "        totaltime += runtime\n"
+    regression_script_code += "        mins = int(np.floor(runtime/60.0))\n"
+    regression_script_code += "        secs = int(np.ceil(runtime - mins*60))\n"
+    regression_script_code += "        print('{:02d}:{:02d} {}'.format(mins, secs, afile))\n"
+    regression_script_code += "        error_out = subprocess.call(['grep', 'Error', outputfile], stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)\n"
```

Comment:
We will also need to grep for "FAIL", since some checks, for example those to see if a test is similar enough to a benchmark, will not include the work "Error" but will say "FAIL".